File: cash.py
from cs50 import get_float
def calculate_quarters(change):
    quarters = change // .25
    return quarters

def calculate_dimes(change):
    dimes = change // .10
    return dimes

def calculate_nickels(change):
    nickels = change // .05
    return nickels

def calculate_pennies(change):
    pennies = change // .01
    return pennies

def main():

    while True:
        change = get_float("Change: ")
        if (change >= 0):
            break

    quarters = calculate_quarters(change)
    change = change - quarters * 0.25

    dimes = calculate_dimes(change)
    change = change - dimes * 0.10

    nickels = calculate_nickels(change)
    changes = change - nickels * 0.05

    pennies = calculate_pennies(change)
    changes = change - pennies * 0.01

    coins = quarters + dimes + nickels + pennies

    print(int(coins))

main()

File: cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int quarters = calculate_quarters(cents);
    cents = cents - (quarters * 25);

    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    int coins = quarters + dimes + nickels + pennies;
    printf("%i\n", coins);
}

int calculate_quarters(int cents)
{
    int quarters = cents / 25;
    return quarters;
}

int calculate_dimes( int cents)
{
    int dimes = cents / 10;
    return dimes;
}
int calculate_nickels(int cents)
{
    int nickels = cents / 5;
    return nickels;
}

int calculate_pennies(int cents)
{
    int pennies = cents;
    return pennies;
}
